src/System.Runtime/tests/System/StringComparerTests.cs

```diff
@@ -5,6 +5,8 @@
 using System.Globalization;
 using Xunit;
 
+#pragma warning disable 0618 // String.Copy is now obsolete
+
```

Comment:
Can we scope this to just where it's used? Should we remove the Copy calls and instead use "new string(s.ToCharArray())" or some such thing?

Also, these comparer examples are interesting. Don't they represent a valid use case, at least for testing purposes?

Reply:
The test itself is valid in that developers who create comparers may want to test a comparer with two objects that have deep equality (if not referential equality). The test will continue to work for the foreseeable future because we're not changing the behavior of `String.Copy` right now - it'll still return a full copy of the string data. The reason for the obsoletion is to prepare for a world where the `String.Copy` API isn't guaranteed to have this behavior due to deduplication.

There are other \[truly horrible\] mechanisms for ensuring that you'll get back a duplicated string, even once `String.Copy` does eventually change to be implemented as `return this;`. But I don't think we'll want to introduce them to these unit tests just yet.